engine/src/main/java/org/terasology/input/lwjgl/LwjglMouseDevice.java

```diff
@@ -96,4 +108,31 @@ public void propertyChange(PropertyChangeEvent evt) {
             this.uiScale = this.renderingConfig.getUiScale() / 100f;
         }
     }
+
+    private void cursorPosCallback(long window, double xpos, double ypos) {
+        xposDelta = xpos - this.xpos;
+        yposDelta = ypos - this.ypos;
+        this.xpos = xpos;
+        this.ypos = ypos;
+    }
+
+    private void mouseButtonCallback(long window, int button, int action, int mods) {
+        ButtonState state;
+        if (action == GLFW.GLFW_PRESS) {
+            state = ButtonState.DOWN;
+        } else if (action == GLFW.GLFW_RELEASE) {
+            state = ButtonState.UP;
+        } else /*if (action == GLFW.GLFW_REPEAT)*/ {
```

Comment:
Mystery commented out code

Reply:
It is some kind a hmm... opaque meaning definition. answer on future question: "Why a else returns ButtonState.REPEAT?"